### PR TITLE
Fix price attribute scope change with app:config:import #33559

### LINKED ISSUE #33559: Problems with price scope and config.php
### Preconditions (*)

1. Magento 2.4.2
2. Atleast 2 store views
3. 1 product which is assigned to atleast 2 store views
4. Price scope is set to global

### Steps to reproduce (*)

1. Set price scope to website in app/etc/config.php

- Initial:

```PHP
'price' => [
     'scope' => null,
],
```

- New:

```PHP
'price' => [
     'scope' => 1,
],
```
2. Execute ``bin/magento app:config:import``
3. Clean magento cache ``bin/magento cache:clean``

### Expected result (*)

1. When I now save different product prices for each website it should save the values for each website

### Actual result (*)

1. The price only gets saved for the Default website

### Solution
The underlying problem is that in the table ``catalog_eav_attribute`` the value for ``is_global`` is '1' (= global scope) for the price attribute if you set the price scope to website via the config.php. This leads to [\Magento\Catalog\Model\Product\Attribute\Backend\Price::Magento\Catalog\Model\Product\Attribute\Backend::afterSave()](https://github.com/magento/magento2/blob/70d1517189ca392c594be3e1c35989ee44103604/app/code/Magento/Catalog/Model/Product/Attribute/Backend/Price.php#L134) not saving the price for each store as it thinks price is a global attribute.

When changing the price scope to website via adminhtml (as seen in the [DevDocs](https://docs.magento.com/user-guide/catalog/catalog-price-scope.html) the column ``is_global``  in the table ``catalog_eav_attribute`` is set to '2' (=website scope) for the price attribute and everything works as supposed.

**So the solution would be to set the ``is_global`` column in the table ``catalog_eav_attribute`` to '2' or '1' for the price attribute when changing the price scope via the config.php file.**

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

The Severity is S1 as now I have to goto every database we have and set the value ``is_global`` in the table ``catalog_eav_attribute`` to '2' for the price attribute, which is something I definitly don't want to do in production environments. I cannot set the price scope via adminhtml as it's blocked as the config value is present in the config.php. So there is a workaround for this but it's wonky as hell.

### PR DESCRIPTION
**Pull request by rmsundar1**
### Description (*)
- With app:config:import command the admin store config change events(e.g _admin_system_config_changed_section_catalog_ ) are not getting triggered, due to this the catalog price attributes scope is not getting change when we update the price scope from config.php.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#33559

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rmsundar1** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@magento run all tests

**Comment by slavvka** [comment]
@magento run all tests
@magento run all tests

**Comment by rmsundar1** [comment]
@magento run  WebAPI Tests, Functional Tests B2B,  Integration Tests

**Comment by rmsundar1** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @rmsundar1,
Could you review and fix failing tests?

File: app/code/Magento/Catalog/Model/Config/PriceScopeChange.php [hunk 1]
```diff
@@ -3,67 +3,54 @@
...
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws \Magento\Framework\Exception\StateException
      *
-     * @param EventObserver $observer
-     * @return void
-     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     * @retrun void
      */
-    public function execute(EventObserver $observer)
+    public function changeScope(int $value)
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/Config/PriceScopeChange.php]:
```suggestion
    public function changeScope(int $value): void
```

File: app/code/Magento/Catalog/Model/Indexer/Product/Price/System/Config/PriceScope.php [hunk 2]
```diff
@@ -31,12 +39,14 @@ public function __construct(
         \Magento\Framework\App\Config\ScopeConfigInterface $config,
         \Magento\Framework\App\Cache\TypeListInterface $cacheTypeList,
         \Magento\Framework\Indexer\IndexerRegistry $indexerRegistry,
+        \Magento\Catalog\Model\Config\PriceScopeChange $priceScopeChange,
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/Model/Indexer/Product/Price/System/Config/PriceScope.php]: Please move it to the last argument  and make it optional with fallback to retrieving it from ObjectManager for backward compatibility reasons

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Model/Indexer/Product/Price/System/Config/PriceScope.php (hunk 2 above)
Comment (rmsundar1) [reply app/code/Magento/Catalog/Model/Indexer/Product/Price/System/Config/PriceScope.php]: Updated the code as per the suggestion.

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @rmsundar1,
Please review and fix the failing tests

**Comment by engcom-Lima** [comment]
@magento run all tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Functional Tests B2B,Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests,WebAPI Tests

**Comment by engcom-Lima** [comment]
@magento run Integration Tests,WebAPI Tests

**Comment by ishakhsuvarov** [comment]
Closing this PR for now as it seems that tests were never fixed for it.

Please feel free to reopen if desired.

**Pull request closed without merging**